DHTExpander/twislave.c

```diff
@@ -79,14 +79,14 @@ ISR (TWI_vect)
 		} else {
 			// Subsequent byte(s) of this transaction
 			// We can now receive data and use it.
+			
 			// We only allow writing the status register 0x00.
 			if(buffer_addr == 0x0)
 			{
-				// And we only allow 0x00 (no DHT power + no readout), 0x02 (no readout + DHT power), and 0x03 (DHT power + readout).
-				if (data == 0x00 || data == 0x02 || data == 0x03) {
-					// Write request to status register
-					i2cdata[buffer_addr]=data;
-				}
+				// We only allow writing to bits 2,3 and setting bits 0,1.
+				i2cdata[buffer_addr] &= 0b11110011; //clear relevant bits except bits 0,1.
```

Comment:
Hmm. This leaves the status register in a weird state for a short time. It's probably fine since this is an interrupt handler, but I'd prefer a copy out of the status register, then modification on that copy, and finally write-back into memory.